src/components/Pressable/GenericPressable/GenericPressable.js

```diff
@@ -0,0 +1,149 @@
+
+const getCursorStyle = (isDisabled, isText) => {
+    if (isDisabled) {
+        return styles.cursorDisabled;
+    }
+    if (isText) {
+        return styles.cursorText;
+    }
+    return styles.cursorPointer;
+};
+
+const GenericPressable = forwardRef((props, ref) => {
+    // eslint-disable-next-line react/destructuring-assignment
+    const {
+        children,
+        onPress,
+        onLongPress,
+        onKeyPress,
+        disabled,
+        style,
+        accessibilityHint,
+        shouldUseHapticsOnLongPress,
+        shouldUseHapticsOnPress,
+        nextFocusRef,
+        keyboardShortcut,
+        shouldUseAutoHitSlop,
+        enableInScreenReaderStates,
+        ...rest
+    } = props;
+
+    const isScreenReaderActive = Accessibility.useScreenReaderStatus();
+    const [hitslop, onLayout] = Accessibility.useAutoHitSlop();
+
+    const isDisabled = useMemo(() => {
+        let shouldBeDisabledByScreenReader = false;
+        switch (enableInScreenReaderStates) {
+            case 'all':
+                shouldBeDisabledByScreenReader = false;
+                break;
+            case 'disabled':
+                shouldBeDisabledByScreenReader = !isScreenReaderActive;
+                break;
+            case 'active':
+                shouldBeDisabledByScreenReader = isScreenReaderActive;
+                break;
+            default:
+                break;
+        }
+        return props.disabled || shouldBeDisabledByScreenReader;
+    }, [isScreenReaderActive, enableInScreenReaderStates, props.disabled]);
+
+    const onLongPressHandler = useCallback(() => {
+        if (!onLongPress) {
+            return;
+        }
+        if (shouldUseHapticsOnLongPress) {
+            HapticFeedback.longPress();
+        }
+        onLongPress();
+        Accessibility.moveAccessibilityFocus(nextFocusRef);
+    }, [shouldUseHapticsOnLongPress, onLongPress, nextFocusRef]);
+
+    const onPressHandler = useCallback(() => {
+        if (shouldUseHapticsOnPress) {
+            HapticFeedback.press();
+        }
+        onPress();
+        Accessibility.moveAccessibilityFocus(nextFocusRef);
+    }, [shouldUseHapticsOnPress, onPress, nextFocusRef]);
+
+    const onKeyPressHandler = useCallback((event) => {
+        if (event.key !== 'Enter') {
+            return;
+        }
+        onPressHandler();
+    }, [onPressHandler]);
+
+    useEffect(() => {
+        if (!keyboardShortcut) {
+            return;
+        }
+        const {shortcutKey, descriptionKey, modifiers} = keyboardShortcut;
+        const unsubscribe = KeyboardShortcut.subscribe(shortcutKey, onPressHandler, descriptionKey, modifiers, true, false, 0, false);
+        return () => {
+            if (!unsubscribe) {
+                return;
+            }
+            unsubscribe();
+        };
+    }, [keyboardShortcut, onPressHandler]);
+
+    return (
+        <Pressable
+            hitSlop={shouldUseAutoHitSlop && hitslop}
```

Comment:
Do you know if there's any time we would use `shuoldUseAutoHitSlop={false}`? I'm wondering if we can just simplify and always use autoHitSlop

Reply:
F.e. link inside the text block. Applying hitslop could break the layout consistency in it.

Reply:
Ok ... my understanding with `hitSlop` is that it would never affect layout and never extend beyond the bounds of the parent. I'll play with this some more and see if I can break the layout of a text block with hitSlop

Reply:
there is another point, maybe more meaningful. The minimal size of the accessible component is stated inside https://www.w3.org/WAI/WCAG21/Understanding/target-size.html says that it can be omited for the inline

Reply:
also two inline links which are close to each other would have overlapping touchableAreas after applying hitslops